app/controllers/concerns/admin/workshop_concerns.rb

```diff
@@ -7,14 +7,19 @@ module Admin::WorkshopConcerns
 
   module InstanceMethods
     def set_admin_workshop_data
-      @attending_students = InvitationPresenter.decorate_collection(@workshop.attending_students.all)
-      @attending_coaches = InvitationPresenter.decorate_collection(@workshop.attending_coaches.all)
-      @coach_waiting_list = WaitingListPresenter.new(WaitingList.by_workshop(@workshop)
-                                                .where_role('Coach')
-                                                .order(:created_at))
-      @student_waiting_list = WaitingListPresenter.new(WaitingList.by_workshop(@workshop)
-                                                  .where_role('Student')
-                                                  .order(:created_at))
+      @attending_students = InvitationPresenter.decorate_collection(
+        @workshop.attending_students.all.with_notes_and_their_authors
+      )
+      @attending_coaches = InvitationPresenter.decorate_collection(
+        @workshop.attending_coaches.all.with_notes_and_their_authors
+      )
+
+      @coach_waiting_list = WaitingListPresenter.new(
+        WaitingList.by_workshop(@workshop).where_role('Coach').order(:created_at).with_notes_and_their_authors
```

Comment:
@gnclmorais not blocking but I'm wondering if it'd be worth moving the code block that retrieves waitlisted coaches and students (on line 21) into their own methods on member, similarly to `attending_coaches` and `attending_students`. I think that would be make it a bit easier to digest this method.

Reply:
I agree, I can move part of it to a new method. Good suggestion, thanks!